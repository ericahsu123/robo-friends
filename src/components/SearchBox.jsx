import React from 'react';
import { connect } from 'react-redux';

import { changeSearchString } from '../actions';

const SearchBox = (props) => {
  const handleChange = (event) => {
    props.changeSearchString(event.target.value);
  };
  return (
    <div className="right item">
      <div className="ui icon input">
        <input
          type="text"
          placeholder="Search robots..."
          onChange={handleChange}
          value={props.query}
        />
        <i className="search icon"></i>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    query: state.query,
  };
};

export default connect(mapStateToProps, { changeSearchString })(SearchBox);
