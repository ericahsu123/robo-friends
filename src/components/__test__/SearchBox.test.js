import React from 'react';
// We're using our own custom render function and not RTL's render
// our custom utils also re-export everything from RTL
// so we can import fireEvent and screen here as well
import { render, cleanup, fireEvent } from './ test-utils';
import '@testing-library/jest-dom/extend-expect';
import SearchBox from '../SearchBox';

afterEach(cleanup);

test('<SearchBox/> component matches snapshot', () => {
  const { container } = render(<SearchBox />, {
    initialState: { query: 'hi' },
  });
  expect(container).toMatchSnapshot();
});

test('Renders the connected <SearchBox/> with initialState', () => {
  const { getByTestId } = render(<SearchBox />, {
    initialState: { query: 'monster' },
  });
  const searchInput = getByTestId('search-input');
  expect(searchInput).toHaveAttribute('value', 'monster');
});

test('Renders the connected <SearchBox/> with input change', () => {
  const { getByTestId } = render(<SearchBox />, {
    initialState: { query: 'monster' },
  });
  const searchInput = getByTestId('search-input');

  fireEvent.change(searchInput, { target: { value: 'hello' } });

  expect(searchInput).toHaveAttribute('value', 'hello');
});
