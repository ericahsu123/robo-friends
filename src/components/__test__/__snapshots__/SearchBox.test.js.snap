// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBox/> component matches snapshot 1`] = `
<div>
  <div
    class="right item"
  >
    <div
      class="ui icon input"
    >
      <input
        aria-label="Search Robots"
        data-testid="search-input"
        placeholder="Search robots..."
        type="text"
        value="hi"
      />
      <i
        class="search icon"
      />
    </div>
  </div>
</div>
`;
